utils/build.ps1

```diff
@@ -2305,31 +2305,49 @@ function Build-ExperimentalRuntime {
   }
 }
 
-function Write-SDKSettingsPlist([Hashtable] $Platform) {
+function Write-SDKSettingsPlist([OS] $OS) {
   $SDKSettings = @{
     DefaultProperties = @{
     }
   }
-  if ($Platform.OS -eq [OS]::Windows) {
+  if ($OS -eq [OS]::Windows) {
     $SDKSettings.DefaultProperties.DEFAULT_USE_RUNTIME = "MD"
   }
-  Write-PList -Settings $SDKSettings -Path "$(Get-SwiftSDK $Platform.OS)\SDKSettings.plist"
+  Write-PList -Settings $SDKSettings -Path "$(Get-SwiftSDK $OS)\SDKSettings.plist"
 
   $SDKSettings = @{
-    CanonicalName = "$($Platform.Triple)"
-    DisplayName = "$($Platform.OS.ToString())"
+    CanonicalName = $OS.ToString()
+    DisplayName = $OS.ToString()
     IsBaseSDK = "NO"
     Version = "${ProductVersion}"
     VersionMap = @{}
     DefaultProperties = @{
-      PLATFORM_NAME = "$($Platform.OS.ToString())"
+      PLATFORM_NAME = $OS.ToString()
       DEFAULT_COMPILER = "${ToolchainIdentifier}"
     }
+    SupportedTargets = @{
+      $OS.ToString() = @{
+        PlatformFamilyDisplayName = $OS.ToString()
+        PlatformFamilyName = $OS.ToString()
+      }
+    }
   }
-  if ($Platform.OS -eq [OS]::Windows) {
-    $SDKSettings.DefaultProperties.DEFAULT_USE_RUNTIME = "MD"
+  switch ($OS) {
+    Windows {
+      $SDKSettings.DefaultProperties.DEFAULT_USE_RUNTIME = "MD"
+      $SDKSettings.SupportedTargets.Windows.LLVMTargetVendor = "unknown"
+      $SDKSettings.SupportedTargets.Windows.LLVMTargetSys = "windows"
+      $SDKSettings.SupportedTargets.Windows.LLVMTargetTripleEnvironment = "msvc"
+      $SDKSettings.SupportedTargets.Windows.Archs = $WindowsSDKPlatforms | ForEach-Object { $_.Architecture.LLVMName } | Sort-Object
+    }
+    Android {
+      $SDKSettings.SupportedTargets.Android.LLVMTargetVendor = "unknown"
+      $SDKSettings.SupportedTargets.Android.LLVMTargetSys = "linux"
+      $SDKSettings.SupportedTargets.Android.LLVMTargetTripleEnvironment = "android${AndroidAPILevel}"
```

Comment:
FWIW, SwiftBuild has android and windows as lowercase under "SupportedTargets".

Environment also should be androideabi for armv7, but we have no way to vary that per architecture right now.

Reply:
Interesting. I thought it was androideabi as well. However, when we recently updated the install rules to use the output of `swiftc -print-target-info`, we end up with `armv7-unknown-linux-android`. I'm happy to change this to androideabi for armv7 though.

Reply:
> I'm happy to change this to androideabi for armv7 though.

You can't at present, because the current schema only allows specifying the triple components per logical platform not per architecture. We'd need to extend the SDKSettings schema (the "authoritative" implementation lives in Swift Build) to allow that.